src/js/index.js

```diff
@@ -0,0 +1,249 @@
+const ADD_TASK_FORM = QS('.todo-list__add');
+
+const OPEN_LIST_CLEAR = QS('.todo-list__open-list .todo-list__clear');
+const DONE_LIST_CLEAR = QS('.todo-list__done-list .todo-list__clear');
+
+const TIME_FORMAT = {
+  hour: '2-digit',
+  minute: '2-digit',
+  hour12: true,
+};
+
+OPEN_LIST_CLEAR.addEventListener('click', clearOpenTasks);
+DONE_LIST_CLEAR.addEventListener('click', clearDoneTasks);
+ADD_TASK_FORM.addEventListener('submit', handleAddSubmit);
+SEARCH_INPUT.addEventListener('input', delayedRenderTasks);
+OPEN_LIST_SORT.addEventListener('change', handleSortChange);
+DONE_LIST_SORT.addEventListener('change', handleSortChange);
+
+document.addEventListener('DOMContentLoaded', () => {
+  initSorts();
+  delayedRenderTasks();
+});
+
+function clearOpenTasks() {
+  TASKS_STORAGE.clearOpenTasks();
+  delayedRenderTasks();
+}
+
+function clearDoneTasks() {
+  TASKS_STORAGE.clearDoneTasks();
+  delayedRenderTasks();
+}
+
+function handleAddSubmit(event) {
+  event.preventDefault();
+  let input = event.target.taskText;
+  let newText = input.value;
+  if (newText) {
+    TASKS_STORAGE.addTask({ taskText: newText });
+    input.value = '';
+    delayedRenderTasks();
+  }
+}
+
+function handleSortChange() {
+  SORT_STORAGE.storeSorts({
+    openSort: OPEN_LIST_SORT.value,
+    doneSort: DONE_LIST_SORT.value,
+  });
+  delayedRenderTasks();
+}
+
+function initSorts() {
+  for (let sortKey of SortStorage.AVAILABLE_SORT_KEYS) {
+    let aSort = SortStorage.AVAILABLE_SORTS[sortKey];
+    let sortStatus = aSort.status;
+
+    let appendTo =
+      sortStatus === Task.STATUS.OPEN
+        ? [OPEN_LIST_SORT]
+        : sortStatus === Task.STATUS.DONE
+        ? [DONE_LIST_SORT]
+        : [OPEN_LIST_SORT, DONE_LIST_SORT];
+
+    createSortOption({
+      key: sortKey,
+      caption: aSort.caption,
+      appendTo: appendTo,
+    });
+  }
+
+  SORT_STORAGE.storeSorts();
+}
+
+function renderSavedTasks() {
+  OPEN_TASK_LIST.innerHTML = '';
+  DONE_TASK_LIST.innerHTML = '';
+
+  let sorts = SORT_STORAGE.loadSorts();
+  OPEN_LIST_SORT.value = sorts.openSort;
+  DONE_LIST_SORT.value = sorts.doneSort;
+
+  let tasks = TASKS_STORAGE.loadTasks({
+    filter: searchFilter,
+  });
+
+  tasks
+    .filter(t => t.isDone())
+    .sort(SortStorage.AVAILABLE_SORTS[sorts.doneSort].sort)
+    .forEach(task => renderTask(task));
+
+  tasks
+    .filter(t => t.isOpen())
+    .sort(SortStorage.AVAILABLE_SORTS[sorts.openSort].sort)
+    .forEach(task => renderTask(task));
+}
+
+function renderTask(task) {
+  const taskNode = document.importNode(TASK_TEMPLATE, true);
+  const qs = taskNode.querySelector.bind(taskNode);
+
+  const listTask = qs('.list__task');
+  const textParagraph = qs('.task__text');
+  const createDate = qs('.task__time--created');
+  const dueDate = qs('.task__time--completed');
+  const textArea = qs('.task__text--input');
+  const checkbox = qs('.task__checkbox');
+  const deleteNode = qs('.task__delete');
```

Comment:
Please, use js-... classnames to indicate elements that are used in js.
This is a bad pattern to rely on layout classnames cause layout can be changed.